lib/app_server.py

```diff
@@ -16,12 +20,24 @@
 from lib.utils import format_process
 from lib.utils import warn_unix_socket
 from test import TestRunGreenlet, TestExecutionError
+from threading import Timer
+
+
+def timeout_handler(server_process, test_timeout):
+    color_stdout("Test timeout of %d secs reached\t" % test_timeout)
+    server_process.kill()
 
 
 def run_server(execs, cwd, server, logfile, retval):
     os.putenv("LISTEN", server.iproto)
     server.process = Popen(execs, stdout=PIPE, stderr=PIPE, cwd=cwd)
-    stdout, stderr = server.process.communicate()
+    test_timeout = Options().args.test_timeout
+    timer = Timer(test_timeout, timeout_handler, [server.process, test_timeout])
+    try:
+        timer.start()
+        stdout, stderr = server.process.communicate()
+    finally:
+        timer.cancel()
```

Comment:
I don't see any reason in try-catch block here. `communicate()` does not raise any exception AFAIK.

Reply:
Removed as not really needed for now.